homeassistant/components/plaato/config_flow.py

```diff
@@ -1,10 +1,227 @@
+        self._errors = {}
+        return await self.async_step_webhook()
+
+    async def async_step_webhook(self, user_input=None):
+        """Validate config step."""
+
+        use_webhook = self._init_info[CONF_USE_WEBHOOK]
+
+        if use_webhook and user_input is None:
+            webhook_id, webhook_url, cloudhook = await self._get_webhook_id()
+            self._init_info[CONF_WEBHOOK_ID] = webhook_id
+            self._init_info[CONF_CLOUDHOOK] = cloudhook
+
+            return self.async_show_form(
+                step_id="webhook",
+                errors=self._errors,
+                description_placeholders={
+                    PLACEHOLDER_WEBHOOK_URL: webhook_url,
+                    PLACEHOLDER_DOCS_URL: DOCS_URL,
+                },
+            )
+
+        return await self._async_step_create_entry()
+
+    async def _async_step_create_entry(self):
+        """Create the entry step."""
+
+        webhook_id = self._init_info.get(CONF_WEBHOOK_ID, None)
+        auth_token = self._init_info[CONF_TOKEN]
+        device_name = self._init_info[CONF_DEVICE_NAME]
+        device_type = self._init_info[CONF_DEVICE_TYPE]
+
+        unique_id = auth_token if auth_token else webhook_id
+
+        await self.async_set_unique_id(unique_id)
+        self._abort_if_unique_id_configured()
+
+        return self.async_create_entry(
+            title=device_type.name,
+            data=self._init_info,
+            description_placeholders={
+                PLACEHOLDER_DEVICE_TYPE: device_type.name,
+                PLACEHOLDER_DEVICE_NAME: device_name,
+            },
+        )
+
+    async def _show_api_method_form(self, device_type: PlaatoDeviceType):
+        data_scheme = vol.Schema({vol.Optional(CONF_TOKEN, default=""): str})
+
+        if device_type == PlaatoDeviceType.Airlock:
+            data_scheme = data_scheme.extend(
+                {vol.Optional(CONF_USE_WEBHOOK, default=False): bool}
+            )
+
+        return self.async_show_form(
+            step_id="api_method",
+            data_schema=data_scheme,
+            errors=self._errors,
+            description_placeholders={PLACEHOLDER_DEVICE_TYPE: device_type.name},
+        )
+
+    async def _get_webhook_id(self):
+        """Generate webhook ID."""
+        webhook_id = self.hass.components.webhook.async_generate_id()
+        if self.hass.components.cloud.async_active_subscription():
+            webhook_url = await self.hass.components.cloud.async_create_cloudhook(
+                webhook_id
+            )
+            cloudhook = True
+        else:
+            webhook_url = self.hass.components.webhook.async_generate_url(webhook_id)
+            cloudhook = False
+
+        return webhook_id, webhook_url, cloudhook
+
+    @staticmethod
+    def _get_error(device_type: PlaatoDeviceType):
+        return (
+            "no_api_method"
+            if device_type == PlaatoDeviceType.Airlock
+            else "no_auth_token"
+        )
```

Comment:
Please switch this to an `if` block.

The standard is to not use a ternary if it spans multiple lines